### PR TITLE
Add Ikigai Mangas

### LINKED ISSUE #19338: Yugen Mangas (Spanish): Changed their domain and name
### Source information

YugenMangas

### Source language

Español 

### Source new URL

ikigaimangas.com

### Other details

IMG_20231216_232404

### Acknowledgements

- [X] I have updated all installed extensions.
- [X] I have opened WebView and checked that the source URL is not updated yet.
- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by bapeey**
Closes #19338 

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [ ] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

File: src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+    private val json: Json by injectLazy()
+
+    private val dateFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS'Z'", Locale.US).apply {
+        timeZone = TimeZone.getTimeZone("UTC")
+    }
+
+    override fun popularMangaRequest(page: Int): Request {
+        val apiUrl = "$apiBaseUrl/api/swf/series?page=$page&column=view_count&direction=desc".toHttpUrl().newBuilder()
+            .build()
```
Comment (AwkwardPeak7) [inline src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt:49]: No need to make it HttpUrl
```suggestion
        val apiUrl = "$apiBaseUrl/api/swf/series?page=$page&column=view_count&direction=desc"
```

File: src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        return GET(apiUrl, headers)
+    }
+
+    override fun popularMangaParse(response: Response): MangasPage = searchMangaParse(response)
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val apiUrl = "$apiBaseUrl/api/swf/series?page=$page&column=last_chapter_date&direction=desc".toHttpUrl().newBuilder()
+            .build()
```
Comment (AwkwardPeak7) [inline src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt:57]:
```suggestion
        val apiUrl = "$apiBaseUrl/api/swf/series?page=$page&column=last_chapter_date&direction=desc"
```

File: src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+            .joinToString(",")
+
+        if (genres.isNotEmpty()) apiUrl.addQueryParameter("genres", genres)
+        if (statuses.isNotEmpty()) apiUrl.addQueryParameter("status", statuses)
+
+        apiUrl.addQueryParameter("column", sortByFilter?.selected ?: "name")
+        apiUrl.addQueryParameter("direction", if (sortByFilter?.state?.ascending == true) "asc" else "desc")
+        apiUrl.addQueryParameter("type", "comic")
+
+        return GET(apiUrl.build().toString(), headers)
```
Comment (AwkwardPeak7) [inline src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt:89]:
```suggestion
        return GET(apiUrl.build(), headers)
```

File: src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+    override fun chapterListRequest(manga: SManga): Request {
+        val id = manga.url.substringAfterLast("#")
+        return GET("$apiBaseUrl/api/swf/series/$id/chapter-list")
+    }
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val result = json.decodeFromString<PayloadChaptersDto>(response.body.string())
+        return result.data.map {
+            SChapter.create().apply {
+                url = "$pageViewerUrl/capitulo/${it.id}"
```
Comment (AwkwardPeak7) [inline src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt:134]: avoid putting full url here, instead add it in `getChapterUrl`

File: src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+        SortProperty("Nombre", "name"),
+        SortProperty("Creado en", "created_at"),
+        SortProperty("Actualización más reciente", "last_chapter_date"),
+        SortProperty("Número de favoritos", "bookmark_count"),
+        SortProperty("Número de valoración", "rating_count"),
+        SortProperty("Número de vistas", "view_count"),
+    )
+
+    override fun getFilterList(): FilterList {
+        runCatching { fetchFilters() }
```
Comment (AwkwardPeak7) [inline src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt:177]: not sure but maybe avoid network calls in `getFilterList`

Comment (vetleledaal) [reply src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt:177]: There is something similar in the Madara class. It stops after 3 failed requests or when it receives the data. Could be worth looking into

Comment (AwkwardPeak7) [reply src/es/ikigaimangas/src/eu/kanade/tachiyomi/extension/es/ikigaimangas/IkigaiMangas.kt:177]: That updates the filters inside `popular/latest/searchparse` methods, and the concept is same here as well. But updating it inside `getFilterList` might be ui blocking?

**Pull request merged**